R/parallel_new.R

```diff
@@ -266,6 +266,11 @@ get_parallel_seeds <- function(workers) {
 #' - If you specify fewer than two workers, or if there is only a single task,
 #'  the computations will occur sequentially.
 #'
+#' Also, we want the results for [last_fit()] to be the same as what one would
+#' get by executing those steps manually (i.e., training the workflow,
+#' predicting the test set, etc.). If [last_fit()] is used, we don’t alter the
+#' random number stream.
```

Comment:
The reference to the random number stream is a lot more technical than the rest of the info here and feels a little out of place to me. Given that  `last_fit()` essentially has only one task (one workflow fitted to one training set), it follows from the 3rd exception above that it's carried out sequentially, so things should be reproducible. 

Talking about what is reproducible and when, is a good thing though! I would just make this its own section, and add a note about running sequentially vs in parallel (and that it doesn't matter if future or mirai), as well as tune 1.3.0 vs new tune.

Reply:
Emphasis added by me: 

> it follows from the 3rd exception above that it's carried out sequentially, so **things should be reproducible**.

The problem was that `last_fit()` wasn't reproducible in a different way. We're fixing that here. 

I've reworded some things, removed references to the random number stream, and added a few more specific exceptions that I could think of.